conda/gateways/connection/adapters/http.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2008-2023 The pip developers
+# SPDX-License-Identifier: MIT
+#
+"""Defines HTTP transport adapter for CondaSession (requests.Session).
+
+Taken from pip (https://github.com/pypa/pip/blob/8c24fd2a80bad21aa29aec02fb48bd89a1e8f5e1/src/pip/_internal/network/session.py#L254
```

Comment:
I think we will need to include https://github.com/pypa/pip/blob/8c24fd2a80bad21aa29aec02fb48bd89a1e8f5e1/LICENSE.txt

Reply:
I copied the whole license file in the header